src/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.cpp

```diff
@@ -0,0 +1,224 @@
+/// Apache License 2.0
+
+#include <numeric>
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+#include <OpenSpaceToolkit/Core/Utility.hpp>
+
+#include <OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.hpp>
+
+namespace ostk
+{
+namespace astrodynamics
+{
+namespace flight
+{
+
+using EarthGravitationalModel = ostk::physics::environment::gravitational::Earth;
+
+const Shared<const Frame> Maneuver::DefaultAccelFrameSPtr = Frame::GCRF();
+
+Maneuver::Maneuver(
+    const Array<Instant>& anInstantArray,
+    const Array<Vector3d>& anAccelerationProfile,
+    const Shared<const Frame>& aFrameSPtr,
+    const Array<Real>& aMassFlowRateProfile
+)
+    : instants_(anInstantArray),
+      accelerationProfileDefaultFrame_(anAccelerationProfile),
+      massFlowRateProfile_(aMassFlowRateProfile)
+{
+    // Sanitize the inputs
+    if (instants_.getSize() != accelerationProfileDefaultFrame_.getSize())
+    {
+        throw ostk::core::error::RuntimeError(
+            "Acceleration profile must have the same number of elements as the number of instants."
+        );
+    }
+
+    if (this->instants_.isEmpty() || this->accelerationProfileDefaultFrame_.isEmpty() ||
+        this->massFlowRateProfile_.isEmpty())
+    {
+        throw ostk::core::error::RuntimeError("No instants or accompanying accelerations/mass flow rates provided.");
+    }
+
+    for (Size k = 0; k < instants_.getSize() - 1; ++k)
+    {
+        if (instants_[k] > instants_[k + 1])
+        {
+            throw ostk::core::error::runtime::Wrong("Unsorted Instant Array");
+        }
+    }
+
+    // Convert to the default frame if necessary
+    accelerationProfileDefaultFrame_ = this->convertAccelerationProfileFrame(aFrameSPtr);
+}
+
+bool Maneuver::operator==(const Maneuver& aManeuver) const
+{
+    return instants_ == aManeuver.instants_ &&
+           accelerationProfileDefaultFrame_ == aManeuver.accelerationProfileDefaultFrame_ &&
+           massFlowRateProfile_ == aManeuver.massFlowRateProfile_;
+}
+
+bool Maneuver::operator!=(const Maneuver& aManeuver) const
+{
+    return !(*this == aManeuver);
+}
+
+std::ostream& operator<<(std::ostream& anOutputStream, const Maneuver& aManeuver)
+{
+    aManeuver.print(anOutputStream);
+
+    return anOutputStream;
+}
+
+bool Maneuver::isDefined() const
+{
+    return true;
+}
```

Comment:
Maybe we should just delete this if it's not being used at all and it's always returning true?

Reply:
I've seen other objects do this as well, but yea that might be a good idea. I'll leave it here for consistency for now thought.